### PR TITLE
[Keras] Add support for element wise comparison operators #4182

### PR DESCRIPTION
**Pull request by shivarajugowda**
Added element wise !=, >, >=, < and <= comparison operators

### CODE REVIEW

File: src/operator/tensor/elemwise_binary_scalar_op_basic.cc [hunk 1]
```diff
@@ -47,6 +47,26 @@ MXNET_OPERATOR_REGISTER_BINARY(_backward_rdiv_scalar)
 MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_equal_scalar)
 .set_attr<FCompute>("FCompute<cpu>", BinaryScalarCompute<cpu, mshadow_op::eq>)
 .add_alias("_EqualScalar");
+    
+MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_not_equal_scalar)
```
Comment (shivarajugowda) [inline src/operator/tensor/elemwise_binary_scalar_op_basic.cc]: OK, I will create separate files for the comparison operators and move the existing "equal" operator and the new comparison operators from the existing files to the new file.

- elemwise_binary_scalar_op_logic.cc
- elemwise_binary_broadcast_op_logic.cc
- elemwise_binary_scalar_op_logic.cu
- elemwise_binary_broadcast_op_logic.cu

File: python/mxnet/ndarray.py [hunk 2]
```diff
@@ -190,6 +190,21 @@ def __rpow__(self, other):
     def __eq__(self, other):
         return equal(self, other)
 
+    def __ne__(self, other):
```
Comment (shivarajugowda) [inline python/mxnet/ndarray.py]: Sure.

File: tests/python/unittest/test_ndarray.py [hunk 3]
```diff
@@ -346,6 +346,50 @@ def test_ndarray_equal():
     z = 0 == x
     assert (z.asnumpy() == np.ones((2, 3))).all()
 
+def test_ndarray_not_equal():
```
Comment (shivarajugowda) [inline tests/python/unittest/test_ndarray.py]: I will keep these around for testing ndarray and add new equivalent ones in testOperator for testing symbol.py

**Comment by shivarajugowda** [comment]
I think I screwed up on the commit, I forgot to rebase my fork to the most recent version of the nnvm. Let me try it again.

**Comment by piiswrong** [comment]
I merged changes from master to nnvm. You need to checkout the upstream branch and cherry pick your changes

**Comment by shivarajugowda** [comment]
Moving this request to #4190 since I had to re-fork and checkin the changes.

**Pull request closed without merging**